Выполненное финальное задание четвертого спринта

stooodgy opened the pull request.

Commit:
Выполненное финальное задание четвертого спринта

AlexKitIt commented on page_object/base_page.py:
```diff
@@ -0,0 +1,28 @@
+    def __init__(self, driver):
+        self.driver = driver
+
+    @allure.step("Нажать на логотип самоката")
+    def click_on_samkat_logo(self):
+        return self.driver.find_element(*BasePagelocators.SAMOKAT_LOGO).click()
+
+    @allure.step("Нажать на логотип Яндекса")
+    def click_on_yandex_logo(self):
+        return self.driver.find_element(*BasePagelocators.YANDEX_LOGO).click()
```
Нужно исправить: в base не может быть методов с конкретными локаторами, это больше класс интерфейс. Сюда еще можно добавить find или click

stooodgy replied:
Исправил - теперь в base только базовые методы

AlexKitIt commented on page_object/main_page.py:
```diff
@@ -0,0 +1,37 @@
+from data_test import QuestionsAnswersDictionary
+
+
+class MainPage:
```
Нужно исправить: Все страницы в POM должны наследоваться от base чтобы переиспользовать все методы base

stooodgy replied:
Сделал через наследование, кода стало меньше и надеюсь он стал более читабельный 
Спасибо за полезную обратную связь

AlexKitIt commented on page_object/main_page.py:
```diff
@@ -0,0 +1,37 @@
+    def click_on_questions_about_important(self, question_index):
+        button = self.driver.find_elements(*MainPagelocators.QUESTION_MENU)
+        button[question_index].click()
+        WebDriverWait(self.driver, 20).until(expected_conditions.visibility_of_element_located((By.XPATH, MainPagelocators.QUESTION_MENU[1])))
```
Нужно исправить: при правильном base переиспользуем  wait_for_load_page(как у тебя) только это будет обычный wait

AlexKitIt commented on tests/test_questions_about_important.py:
```diff
@@ -0,0 +1,18 @@
+
+class TestQestionsAboutImportant:
+
+    @pytest.mark.parametrize('question_index', [0, 1, 2, 3, 4, 5, 6, 7])
```
Хороший вариант

Commit:
Исправление замечаний полученных в ходе code review:
1) Из base_page убраны конкретные локаторы, добавлены общие методы для взаимодействия с элементами страниц.
2) main_page и order_page наследуются от base_page и переиспользуют его методы

AlexKitIt commented on page_object/base_page.py:
```diff
@@ -0,0 +1,38 @@
+from selenium.webdriver.support import expected_conditions
+
+
+class BasePage:
```
Вот это хороший Base

AlexKitIt commented on page_object/main_page.py:
```diff
@@ -0,0 +1,42 @@
+        self.click_on(MainPagelocators.BUTTON_ACCEPT_COOKIES)
+
+    @allure.step("Нажать на кнопку заказа")
+    def click_order_button(self, button = MainPagelocators.BUTTON_ORDER_TOP):
```
Лайк за значение аргумента по умолчанию
